### PR TITLE
Fix sync timeout

### PR DESCRIPTION
**Pull request by fbernitt**
The sequential handling of downloading and inserting docs blocked the twisted event loop (see the _process_decrypted_docs in the diff below)

```python
         for doc_fields in insertable:
             self._insert_decrypted_local_doc(*doc_fields)
```

Move the insertion into a background thread to keep the event loop free and able to handle the download requests

### CODE REVIEW

File: client/src/leap/soledad/client/encdecpool.py [hunk 1]
```diff
@@ -687,7 +688,8 @@ def _process_decrypted_docs(self):
         """
         insertable = yield self._get_insertable_docs()
         for doc_fields in insertable:
-            self._insert_decrypted_local_doc(*doc_fields)
+            method = self._insert_decrypted_local_doc
+            yield threads.deferToThread(method, *doc_fields)
```
Comment (meskio) [inline client/src/leap/soledad/client/encdecpool.py]: Can you add a comment here mentioning why was this needed?

**Comment by shyba** [comment]
@meskio fixed, thanks for reviewing.
